src/api/auth/gg-profile.js

```diff
@@ -0,0 +1,78 @@
+import axios from 'axios';
+import connect from 'next-connect';
+import { authInstance } from '../../utilities/authInstance';
+import { ExceptionHandler } from '../../utilities/ApiExceptions';
+
+/**
+ * Get Bearer Token
+ * Fetches and returns a bearer token string using the client/secret provided.
+ */
+const baseUrl = process.env.MEGAPROFILE_TOKEN_URL;
+const params = {
+  client_id: process.env.MEGAPROFILE_CLIENT_ID,
+  client_secret: process.env.MEGAPROFILE_CLIENT_SECRET,
+  grant_type: 'client_credentials',
+};
+const tokenFetcher = async () => {
+  const response = await axios.post(baseUrl, null, { params });
+
+  if (response && response.data && response.data.access_token) {
+    return response.data.access_token;
+  }
+
+  console.error(response);
+  throw new Error('Response did not contain access token');
+};
+
+/**
+ * Get profile data from the MEGA PROFILE API
+ */
+
+const profileFetcher = async (profileID, token) => {
+  let response;
+  const endpoint = `${process.env.MEGAPROFILE_PROFILE_URL}/${profileID}/${process.env.MEGAPROFILE_PROFILE_TYPE}`;
```

Comment:
You won't have to make this configurable with the changes to the API structures. You can remove the profile type environment var.

```suggestion
  const endpoint = `${process.env.MEGAPROFILE_PROFILE_URL}/${profileID}/fullgg`;
```